src/js/jsx/search/SearchBar.jsx

```diff
@@ -416,15 +427,19 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onChange}
+         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
+         * @param {string} searchTerm user input
+         * @return {{dontCloseDialog: boolean=}};
          */
-        _handleChange: function (id) {
-            if (id === null) {
+        _handleChange: function (id, searchTerm) {
+            if (!id) {
                 this.props.dismissDialog();
                 return;
             }
 
-            var filters = this.state.filter,
+            var filters = this.state.filters,
+                lastFilter = filters ? filters[filters.length - 1] : id,
```

Comment:
Er, actually, it will just be undefined, because `filters.length - 1` will be -1, and `[-1] === undefined`. Anyway, Is this what you want?